src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberFieldSymbol.cs

```diff
@@ -49,15 +49,15 @@ protected void TypeChecks(TypeSymbol type, DiagnosticBag diagnostics)
             }
             else if (type.SpecialType == SpecialType.System_Void)
             {
-                diagnostics.Add(ErrorCode.ERR_FieldCantHaveVoidType, TypeSyntax.Location);
+                diagnostics.Add(ErrorCode.ERR_FieldCantHaveVoidType, TypeSyntax?.Location ?? this.Locations[0]);
             }
             else if (type.IsRestrictedType(ignoreSpanLikeTypes: true))
             {
-                diagnostics.Add(ErrorCode.ERR_FieldCantBeRefAny, TypeSyntax.Location, type);
+                diagnostics.Add(ErrorCode.ERR_FieldCantBeRefAny, TypeSyntax?.Location ?? this.Locations[0], type);
             }
             else if (type.IsByRefLikeType && (this.IsStatic || !containingType.IsByRefLikeType))
             {
-                diagnostics.Add(ErrorCode.ERR_FieldAutoPropCantBeByRefLike, TypeSyntax.Location, type);
+                diagnostics.Add(ErrorCode.ERR_FieldAutoPropCantBeByRefLike, TypeSyntax?.Location ?? this.Locations[0], type);
```

Comment:
>diagnostics.Add(ErrorCode.ERR_FieldAutoPropCantBeByRefLike, TypeSyntax?.Location ?? this.Locations[0], type); [](start = 16, length = 109)

Are there tests that hit modified code paths when ```TypeSyntax``` is null?

Reply:
Yes.

---
In reply to: [156531308](https://github.com/dotnet/roslyn/pull/23209#discussion_r156531308) [](ancestors = 156531308)